taichi/backends/opengl/opengl_api.h

```diff
@@ -47,8 +51,7 @@ struct CompiledProgram {
 
   void add(const std::string &kernel_name,
            const std::string &kernel_source_code,
-           int num_groups,
-           RangeSizeEvaluator rse,
+           KernelParallelAttrib kpa,
```

Comment:
nit: If you want to make sure the caller uses `std::move`, you can change the type to `KernelParallelAttrib&&`

Reply:
Cool C++11! Thank for telling me that, I finally realized the difference between `&` and `&&`: `&&` must be moved lhs, while `&` only need to be lhs.